### PR TITLE
Adds FreeRTOS v11 and a skeleton Rust app.

### PR DESCRIPTION
**Pull request by jonathanpallant**
Based on the threadx-experiments repo.

### CODE REVIEW

**Comment by jonathanpallant** [comment]
I'll squash this once it builds OK. I also want to add some timers to show the task switching is working.

**Comment by jonathanpallant** [comment]
Added a second thread, and a shared queue which is heap allocated.

```console
$ DEFMT_LOG=debug cargo run --release
   Compiling freertos-sys v0.1.0 (/Users/jonathan/Documents/ferrous-systems/freertos-experiments/freertos-sys)
   Compiling demo-app v0.0.0 (/Users/jonathan/Documents/ferrous-systems/freertos-experiments/demo-app)
    Finished release [optimized + debuginfo] target(s) in 1.78s
     Running `probe-rs run --chip nRF52840_xxAA target/thumbv7em-none-eabihf/release/demo-app`
      Erasing ✔ [00:00:00] [############################################################################################################################] 16.00 KiB/16.00 KiB @ 33.31 KiB/s (eta 0s )
  Programming ✔ [00:00:00] [############################################################################################################################] 16.00 KiB/16.00 KiB @ 43.25 KiB/s (eta 0s )    Finished in 0.871s
0 INFO  Hello, this is version unknown!
└─ demo_app::__cortex_m_rt_main @ src/main.rs:23  
0 INFO  Entering FreeRTOS kernel...
└─ demo_app::__cortex_m_rt_main @ src/main.rs:97  
0 INFO  params = 0x2002fb30
└─ demo_app::test_task2 @ src/main.rs:131 
0 INFO  params = 0x2002fb30
└─ demo_app::test_task1 @ src/main.rs:109 
0 DEBUG counter = 0
└─ demo_app::test_task1 @ src/main.rs:113 
0 INFO  Got value in queue: 0
└─ demo_app::test_task2 @ src/main.rs:144 
100 DEBUG counter = 1
└─ demo_app::test_task1 @ src/main.rs:113 
100 INFO  Got value in queue: 1
└─ demo_app::test_task2 @ src/main.rs:144 
200 DEBUG counter = 2
└─ demo_app::test_task1 @ src/main.rs:113 
200 INFO  Got value in queue: 2
└─ demo_app::test_task2 @ src/main.rs:144 
```

File: README.md [hunk 1]
```diff
@@ -2,6 +2,32 @@
...
+0 INFO  test_task2(params = 0x0, counter = 0)
+└─ demo_app::test_task2 @ src/main.rs:105 
+0 INFO  test_task1(params = 0x0, counter = 0)
+└─ demo_app::test_task1 @ src/main.rs:95  
+50 INFO  test_task1(params = 0x0, counter = 1)
+└─ demo_app::test_task1 @ src/main.rs:95  
+100 INFO  test_task1(params = 0x0, counter = 2)
+└─ demo_app::test_task1 @ src/main.rs:95  
+```
+
```
Comment (miguelraz) [inline README.md:30]:
```suggestion
Note: Windows users will need an install of [gcc-arm-none-eabi](https://developer.arm.com/Tools%20and%20Software/GNU%20Toolchain).
```

**Review by miguelraz — APPROVED** [review]
This did what it said it had to do in the box - approved!

I've added a suggestion for Windows users to add the compiler I needed.

File: README.md (hunk 1 above)
Comment (jonathanpallant) [reply README.md:30]: I expanded a little bit because that link is also useful for macOS users.

**Pull request merged**